app/models/concerns/assign_to_current_semester.rb

```diff
@@ -6,6 +6,7 @@ module AssignToCurrentSemester
   end
 
   def assign_to_current_semester
-    self.semester = Semester.current_semester unless semester.present?
+    # TODO: Settings are not initilized yet here
+    # self.semester = Semester.find(Settings.new.current_semester_id)
```

Comment:
Right. Is that a valid assumption? If so then we can just raise an error if we get here and the current semester hasn't been set yet.

Alternatively we can put a validation on the Settings model to ensure that a current semester is always set, and then `self.semester = Settings.instance.current_semester` would work.